Objective-C/Tests/URLEndpointListenerTest.m

```diff
@@ -437,4 +437,202 @@ - (void) testUnavailableNetworkInterface {
+    
+    // save blob on db#2
+    [self deleteDBNamed: @"db2" error: &err];
+    AssertNil(err);
+    CBLDatabase* db2 = [self openDBNamed: @"db2" error: &err];
+    AssertNil(err);
+    CBLMutableDocument* doc2 = [self createDocument: @"doc-2"];
+    CBLBlob* blob2 = [[CBLBlob alloc] initWithContentType: @"text/plain" data: content];
+    [doc2 setValue: blob2 forKey: @"blob"];
+    [db2 saveDocument: doc2 error: &err];
+    AssertNil(err);
+    
+    // replicator#1
+    CBLReplicatorConfiguration* config1 = [self configWithTarget: _listener.localEndpoint
+                                                            type: kCBLReplicatorTypePushAndPull
+                                                      continuous: NO];
+    config1.pinnedServerCertificate = (__bridge SecCertificateRef) _listener.config.tlsIdentity.certs[0];
+    CBLReplicator* repl1 = [[CBLReplicator alloc] initWithConfig: config1];
+    
+    // replicator#2
+    CBLReplicatorConfiguration* config2 = [[CBLReplicatorConfiguration alloc] initWithDatabase: db2
+                                                                                        target: _listener.localEndpoint];
+    config2.pinnedServerCertificate = (__bridge SecCertificateRef) _listener.config.tlsIdentity.certs[0];
+    CBLReplicator* repl2 = [[CBLReplicator alloc] initWithConfig: config2];
+    
+    // get listener status
+    __block Listener* weakListener = _listener;
+    __block uint64_t maxConnectionCount = 0, maxActiveCount = 0;
+    id changeListener = ^(CBLReplicatorChange * change) {
+        Listener* listener = weakListener;
+        if (change.status.activity == kCBLReplicatorBusy) {
+            maxConnectionCount = MAX(listener.status.connectionCount, maxConnectionCount);
+            maxActiveCount = MAX(listener.status.activeConnectionCount, maxActiveCount);
+        } else if (change.status.activity == kCBLReplicatorStopped) {
+            if (change.replicator == repl1)
+                [exp1 fulfill];
+            else
+                [exp2 fulfill];
+        }
+    };
+    id token1 = [repl1 addChangeListener: changeListener];
+    id token2 = [repl2 addChangeListener: changeListener];
+    
+    // start & wait for replication
+    [repl1 start];
+    [repl2 start];
+    [self waitForExpectations: @[exp1, exp2] timeout: 5];
+    
+    // check both replicators access listener at same time
+    AssertEqual(maxConnectionCount, 2u);
+    AssertEqual(maxActiveCount, 2u);
+    
+    // all data are transferred to/from
+    AssertEqual(self.otherDB.count, 3u);
+    AssertEqual(self.db.count, 2u);
+    AssertEqual(db2.count, 2u);
+    
+    // cleanup
+    [_listener stop];
+    [repl1 removeChangeListenerWithToken: token1];
+    [repl2 removeChangeListenerWithToken: token2];
+    repl1 = nil;
+    repl2 = nil;
+    Assert([db2 delete: &err]);
+    AssertNil(err);
+    db2 = nil;
+    [self deleteDBNamed: @"db2" error: &err];
+}
+
+- (void) testMultipleReplicatorsOnReadOnlyListener {
```

Comment:
What is the goal of this test? If the goal to test pull and push-and-pull replicator, I personally like to make it into two separate this so that it's simpler and easier to understand. Up to you though.

Reply:
goal is to test multiple replicators simultaneously pull from read only listener.

Reply:
ok